features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/ImageWireframeHelper.kt

```diff
@@ -37,13 +40,14 @@ internal class ImageWireframeHelper(
         prefix: String = DRAWABLE_CHILD_NAME
     ): MobileSegment.Wireframe.ImageWireframe? {
         val id = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(view, prefix + currentWireframeIndex)
+        val (currentDrawable, drawableWidth, drawableHeight) = resolveDrawableProperties(view, drawable)
```

Comment:
Why unbox the components here when. you have an object storing the info? You could then remove the `@Suppress("ComplexCondition")` below by having a `DrawableProperties.isValid(): Boolean` method

Reply:
@xgouchet ripple drawable are by default used as backgrounds for a View (button, text, etc).

Reply:
@xgouchet regarding whether we should be rendering rippledrawable as imagewireframes, I understand you discussed with marius - we have a problem getting the color of the drawable in order to use it in a shape, so need to render the bitmap (internally the rippledrawables are using gradientdrawable)